File: features/lead/src/main/java/com/nivasafinance/features/lead/dto/LeadDocumentResponse.java
package com.nivasafinance.features.lead.dto;

import com.nivasafinance.features.master.codemaster.dto.CodeValueResponse;
import com.nivasafinance.features.master.codemaster.dto.MasterCodeResponse;
import jdk.jfr.Category;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LeadDocumentResponse {
    private UUID identifier;
    private String name;
    private String type;
    private Long size;
    private List<MasterCodeResponse> categories;
    private List<CodeValueResponse> tags;
    private LocalDateTime createdAt;
    private String createdBy;
}

File: features/lead/src/main/java/com/nivasafinance/features/lead/dto/LeadDocumentResponse.kt
package com.nivasafinance.features.lead.dto

import com.nivasafinance.features.document.dto.DocumentResponse
import java.util.UUID

/**
 * Lead-specific document response that wraps the core DocumentResponse
 * with additional lead context information
 */
data class LeadDocumentResponse(
    val taskId: UUID? = null,
    val document: DocumentResponse,
)
